### PR TITLE
=doc #18857 Add docs about comparing the secret when authenticating

### PR DESCRIPTION
**Pull request by johanandren**
Refs #18857

### CODE REVIEW

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/extractCredentials.rst [hunk 1]
```diff
@@ -15,6 +15,8 @@ Description
 Extracts the potentially present ``HttpCredentials`` provided with the request's ``Authorization`` header,
 which can be then used to implement some custom authentication or authorization logic.
 
+See :ref:`credentials-and-timing-attacks-scala` for details about verifying the secret.
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/extractCredentials.rst]: Do we have it mentioned anywhere in the java side of docs? Is there a corresponding API?

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst [hunk 2]
```diff
@@ -64,3 +64,18 @@ they are only a means of extracting the so called ``Bearer Token`` from the ``Au
...
 
 .. _RFC 6750: https://tools.ietf.org/html/rfc6750
+
+
+.. _credentials-and-timing-attacks-scala:
+
+Credentials and password timing attacks
+---------------------------------------
+
+When transforming request `Credentials` into an application specific user identifier the naive solution for
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst]: RST syntax is:

```
`Credentials`
```

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst [hunk 3]
```diff
@@ -64,3 +64,18 @@ they are only a means of extracting the so called ``Bearer Token`` from the ``Au
...
+checking the secret (password) would be a regular string comparison, but doing this would open up the application to
+timing attacks. See for example http://emerose.com/timing-attacks-explained for an explanation of the problem.
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst]: Does this become a link automagically?

**Comment by ktoso** [comment]
LGTM, after small fixup on syntax

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst (hunk 3 above)
Comment (johanandren) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst]: Yup.

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst (hunk 2 above)
Comment (johanandren) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst]: Argh. Damn you multiple markup languages.

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst (hunk 3 above)
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst]: cool :)

Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/index.rst]: Having that said... maybe do make it have a title; we usually do links as `Blabla`_ AFAIR. "It is mroe explained in [nice human readable title, not url]"

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/extractCredentials.rst (hunk 1 above)
Comment (johanandren) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/extractCredentials.rst]: No corresponding API as far as I can see

Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/security-directives/extractCredentials.rst]: That's bad, please add a ticket so we add api or explain how to do it in java

**Comment by ktoso** [comment]
LGTM, thanks - can you please create  ticket about adding API or docs about it to JavaDSL? @johanandren

**Pull request merged**